src/datasources/cache/cache.router.ts

```diff
@@ -408,7 +409,8 @@ export class CacheRouter {
   }
 
   static getRelayKey(args: { chainId: string; address: string }): string {
-    return `${args.chainId}_${CacheRouter.RELAY_KEY}_${args.address}`;
+    // Ensure address is checksummed to always have a consistent cache key
+    return `${args.chainId}_${CacheRouter.RELAY_KEY}_${getAddress(args.address)}`;
```

Comment:
I'm not happy with this being located here but as the cache could theorertically be accessed from any module, I think it makes the most sense. Please let me know your thoughts.

Reply:
I'm more in favor of calling viem's `getAddress` on the domain, not here. I'd keep this class as clean as possible regarding dependencies. Otherwise we could also consider this check can applied to other functions in the this class handling eth addresses. I think we should assume the data coming from the domain is already validated/sanitized.

But besides having said that, it is not a strong opinion, I'm happy to see what you guys think @iamacook @fmrsabino

Reply:
I have moved the checksumming to the domain in a wrapper function in dc73902b4838bcb57f7a97a74f997cc572677a30.